packages/contracts/contracts/rbac.clar

```diff
@@ -0,0 +1,169 @@
+(define-constant CONTRACT_OWNER tx-sender)
+
+;; Stores roles granted to users
+;; Each user can have up to 128 roles ranging from 0-127.
+;; Roles are stored as bitmask values represented as uint's.
+(define-map UserRoles
```

Comment:
Should we just use `(define-map UserRoles principal uint)`

Reply:
Without tuple key names?! I don't recall this as valid syntax.

At this stage I don't know if being less explicit will be beneficial for us or not.
It might reduce execution costs so it is something we need to test.

BTW. If this is a valid syntax, why I can't find it in official documentation?

Reply:
@LNow this is actually how maps are defined, through tuples. it's defined in the docs, see!

<img width="500" alt="Screenshot 2021-04-27 at 17 40 30" src="https://user-images.githubusercontent.com/67651251/116279733-ca33a700-a77f-11eb-9899-2cea93a52b5e.png">

Reply:
Yes, documentation says that you have to supply at minimum `key-name-0`, `key-type-0`, `val-name-0`and `val-type-0`.
If you look again at @friedger suggestion, you'll find only `key-type-0` and `val-type-0`. No names at all.

In theory `map-get?` should return `(optional (tuple))`. But with his approach it will return `(optional (uint))`. :scream:

Reply:
>  But with his approach it will return (optional (uint))
Correct.

The documentation is not complete, it is allowed to use any type as key

Reply:
I will open a separate discussion about this topic.
It is very tempting to use this simplified syntax right away, but I think it will be better to have everyone on the same page why we are doing so, what are the pros and cons of it etc.